coalib/bearlib/languages/definitions/Haskell.py

```diff
@@ -0,0 +1,19 @@
+from coalib.bearlib.languages.Language import Language
+
+
+@Language
+class Haskell:
+    extensions = '.hs',
+    versions = 1.0, 1.1, 1.2, 1.3, 1.4, 98, 2010
+    special_chars = list('!#$%&*+./<=>?@\\^|-~')
+    comment_delimiters = '--',
+    multiline_comment_delimiters = {'{-': '-}'}
+    keywords = [
+        '-<', '-<<', '::', ';', '<-', ',', '_', '{, }',
+        'case', 'of', 'class', 'infixl', 'foreign',
+        'data', 'data family', 'data instance', 'default',
+        'deriving', 'riving instance', 'do', 'forall',
```

Comment:
`deriving` and `instance` already exists. I think you don't need to write it again as `deriving instance`. It is also a typo by the way.